File: Algorithms/283. Move Zeroes/javascript/solution1.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    const n = nums.length;
    let numZeroes = 0;
    for (let i = 0; i < n; i++) {
        numZeroes += nums[i] === 0 ? 1 : 0;
    }

    const ans = [];
    for (const num of nums) {
        if (num !== 0) {
            ans.push(num);
        }
    }
    while (numZeroes--) {
        ans.push(0);
    }
    nums.splice(0, nums.length, ...ans);
};

File: Algorithms/283. Move Zeroes/python/solution1.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        num_zeroes = 0
        for i in range(n):
            num_zeroes += nums[i] == 0

        ans = list()
        for num in nums:
            if num != 0:
                ans.append(num)
        while num_zeroes > 0:
            ans.append(0)
            num_zeroes -= 1
        for i in range(n):
            nums[i] = ans[i]
